### DISCUSSION TITLE
Can't compile anymore, implicit using doesn't seem to work.

### DISCUSSION THREAD

**Discussion by lucapivato** [dotnet/winforms#5408] [Q&A] [answered]
Since the last update we cannot compile winforms and winforms-designer anymore. In both cases the SDK version 6.0.100-rc.1.21379.2 cannot be found and cannot be downloaded. The latest we can find is 6.0.100-preview.6.21355.2.

"using System" is missing and the implicit using in csproj is missing as well causing thousands of compiler error.

Tried with VS2019 and 2022 and on clean dev machines.

What am I missing?

**Comment by RussKie** [comment, marked as answer]
> In both cases the SDK version 6.0.100-rc.1.21379.2 cannot be found and cannot be downloaded.

How are you trying to download it?
Works on my machine (c): https://dotnetcli.azureedge.net/dotnet/Sdk/6.0.100-rc.1.21379.2/dotnet-sdk-6.0.100-rc.1.21379.2-win-x64.zip

> "using System" is missing and the implicit using in csproj is missing as well causing thousands of compiler error.

This is because we're using the new "implicit global usings" introduced in Preview7: https://github.com/dotnet/sdk/pull/18459. This feature is not compatible with VS2019, and only supported in VS2022 and .NET 6.0 Preview7.
However this has been recognised as an intrusive change, and there will be additional follow up work tracked under https://github.com/dotnet/sdk/issues/19521.

**Comment by lucapivato** [comment]
Thank you it works. I had the same link but I was using restore.cmd. I didn't think of downloading it directly.

![image](https://user-images.githubusercontent.com/37640802/128869056-a56601c0-33bc-4079-ba68-8ca8f316bd6f.png)

**Reply by RussKie** [reply]
It looks like your issue is 'install' rather 'download'... Do you have an antivirus or corporate policies that may be interfering?

**Reply by lucapivato** [reply]
Yes I noticed that too after I wrote the question. There is no antivirus other than window defender and no corporate policy. Tried the same on 2 different dev machines. It always worked before.

**Comment by RussKie** [comment]
As an experiment, try the latest version available from dotnet/installer: .NET SDK Installer (github.com)<https://github.com/dotnet/installer>. Update the versions in global.json and try .\restrore.cmd.

**Reply by lucapivato** [reply]
Simply ran build from the /installer repo and the same sdk installation worked:

![image](https://user-images.githubusercontent.com/37640802/128881090-290aaae8-9068-4ce7-8653-c83ea92398b6.png)

Deleted the 6.0 sdk, tried again the restore from /winforms and it worked, it installed 6.0.100-rc.1.21379.2 from the same location as before, then it installed 6.0.0-rc.1.21406.5 (but it's nowhere, not in /dotnet and not in /winforms/.dotnet):

![image](https://user-images.githubusercontent.com/37640802/128884177-dffb7eec-5956-44d8-9ea8-2425f40b98ee.png)

Anyway, it's all working now and another developer here has been able to recompile again correctly. Thank you for your help.

**Reply by RussKie** [reply]
Great! It could be that there was an outage of some kind (I'm seeing some internal comms about various outages)... 
¯\\_(ツ)\_/¯
